src/DataManagement/Goals/Goal.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using ePiggyWeb.DataBase.Models;
+using ePiggyWeb.Utilities;
+
+namespace ePiggyWeb.DataManagement.Goals
+{
+    public class Goal : IGoal
+    {
+        public int Id { get; set; }
+        public int UserId { get; set; }
+        public string Title { get; set; }
+        public decimal Amount { get; set; }
+
+        public Goal(string title, decimal amount)
+        {
+            Title = title;
+            Amount = amount;
+        }
+
+        public Goal(int id, int userId, string title, decimal amount) : this(title, amount)
+        {
+            Id = id;
+            UserId = userId;
+        }
+
+        public Goal(IGoalModel dbGoalModel) : this(dbGoalModel.Id, dbGoalModel.UserId, dbGoalModel.Title, dbGoalModel.Price) { }
+
+        public Goal(int id, int userId, IGoal goal) : this(id, userId, goal.Title, goal.Amount) { }
+
+        public Goal()
+        {
+            Id = 0;
+            UserId = 0;
+            Title = "unnamed";
+            Amount = 0;
+        }
+
+        public void Edit(IGoal goal)
+        {
+            Title = goal.Title;
+            Amount = goal.Amount;
+        }
+
+        // This needs to be amended
+
+        public Goal(string title)
+        {
+            SetGoalFromWeb(title);
+        }
+
+        private static readonly string ResourceDirectoryParsedGoal = Directory.GetParent(Environment.CurrentDirectory)
+                                                                         .Parent?.Parent?.FullName +
+                                                                     @"\resources\textData\parsedGoal.txt";
+        public void SetGoalFromWeb(string itemName)
+        {
+            try
+            {
+                Task.Run(() => InternetParser.ReadPriceFromCamel(itemName)).Wait();
```

Comment:
Thanks for the reminder! I marked it down as a comment, and also commented out the whole section related to reading goals from web, since I don't have the knowledge to port it properly yet into our new project